iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TieLineAdderProxy.java

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.impl;
+
+import com.powsybl.iidm.network.*;
+import java.util.Objects;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ *
+ * @deprecated Use {@link TieLineAdder#newHalfLine1()} or {@link TieLineAdder#newHalfLine2()} instead.
+ */
+@Deprecated
+class TieLineAdderProxy implements TieLineAdder, Validable {
```

Comment:
The proxy should be moved in the API and used in the old API methods. So that it will be really easy to remove existing code in the implementation without breaking existing code.